main/views.py

```diff
@@ -1,7 +1,26 @@
 from django.shortcuts import render
+from django.contrib.auth.views import LoginView
+from django.contrib.auth import authenticate, login
+from django.contrib import messages
 
 
 def homePage(request):
     if request.user.is_authenticated:
         return render(request, 'main/loggedInPage.html')
     return render(request, 'main/homePage.html')
+
+
+class CustomLoginView(LoginView):
+    template_name = 'main/login.html'
+
+    def post(self, request):
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return render(request, 'main/loggedInPage.html')
+        else:
+            error_message = "Invalid username or password."
+            messages.error(request, error_message)
+            return render(request, 'main/login.html')
```

Comment:
If I use what you suggested, the error messages will not appear if the user is not logged in.
The view function is designed so that if the user is connected, it will move the user to the other page and if not an error message will appears to the user.
And will change the LOGIN_URL='/admin' in HighTeach/settings.py to LOGIN_URL='/login'.

Reply:
Makes sense. Don’t worry about it then

Reply:
Done, Updated the LOGIN_UR.